File: core/tests/scripts/lua/fib.lua
local function fib(n)
        if n < 0 then
                return -1
        end

        if n == 0 then
                return 0
        elseif n == 1 then
                return 1
        end

        local n0, n1 = 0, 1

        for i = 2, n, 1 do
                local tmp = n0 + n1
                n0 = n1
                n1 = tmp
        end

        return n1
end

function Test(n)
        return fib(n)
end

File: core/tests/scripts/python/fib.py
def fib(n):

        if n < 0:
                return -1

        if n == 0:
                return 0
        elif n == 1:
                return 1

        n0, n1 = 0, 1

        for _ in range(2, n+1):
                tmp = n0 + n1
                n0 = n1
                n1 = tmp

        return n1

